### PR TITLE
changefeed (ticdc): make changefeed list show warning message report by sinkmanager

### LINKED ISSUE #9594: TiCDC list changefeed API should return warning message when changefeed encounter an warning reported by sinkManager
### What did you do?

1. create a changefeed 
2. inject an error into sinkManager

### What did you expect to see?

Excute `./cdc changefeed list` could see the changefeed was in warning state and can saw the warning message.

### What did you see instead?

Changefeed is in normal state and no error message.

### Versions of the cluster

Upstream TiDB cluster version (execute `SELECT tidb_version();` in a MySQL client):

```console
(paste TiDB cluster version here)
```

Upstream TiKV version (execute `tikv-server --version`):

```console
(paste TiKV version here)
```

TiCDC version (execute `cdc version`):

```console
master
```

### PR DESCRIPTION
**Pull request by asddongmen**
### What problem does this PR solve?

Issue Number: close #9594 

### What is changed and how it works?

Ensure that the changefeed remains in the warning state if it encounter an warning and its checkpoint does not proceed. This will allow the list changefeed API to display the error message for a warning changefeed.

#### Tests 
 
 - unit test
 - Integration test
 - Manual test (add detailed scripts or steps below)

### Release note 

```release-note
make changefeed list show warning message report by sinkManager
```

### CODE REVIEW

**Review by nongfushanquan — APPROVED** [review]

File: cdc/owner/feed_state_manager.go [hunk 1]
```diff
@@ -590,6 +591,12 @@ func (m *feedStateManager) handleWarning(errs ...*model.RunningError) {
 	if m.state.Status != nil {
 		currTime := m.upstream.PDClock.CurrentTime()
 		ckptTime := oracle.GetTimeFromTS(m.state.Status.CheckpointTs)
+		m.lastWarningReportCheckpointTs = m.state.Status.CheckpointTs
```
Comment (hicqu) [inline cdc/owner/feed_state_manager.go:594]: It should be moved into the `if` block?

**Review by hicqu — APPROVED** [review]

**Pull request merged**